All .gliffy files replaced by .svg

d-stahl-ericsson opened the pull request:
### Applicable Issues
#198.

### Description of the Change
Replaced all .gliffy files with equivalent .svg files.

### Alternate Designs
There are multiple vector graphics formats out there, but SVG is a widely adopted standard with plenty of tool support.

### Benefits
Free from dependence on Gliffy as a tool.

### Possible Drawbacks
Any reliance on features unique to the .gliffy format would be lost. I haven't detected any such problems, however.

### Sign-off
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Daniel Ståhl daniel.stahl@ericsson.com

e-backmark-ericsson commented on usage-examples/build-avoidance.svg:
```diff
@@ -0,0 +1,2 @@
+<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
+<svg xmlns="http://www.w3.org/2000/svg" style="background-color: rgb(255, 255, 255);" xmlns:xlink="http://www.w3.org/1999/xlink" width="999px" height="710px" version="1.1"><defs><linearGradient x1="0%" y1="100%" x2="0%" y2="0%" id="mx-gradient-ffffff-1-b4a7d6-1-s-0"><stop offset="0%" style="stop-color:#b4a7d6"/><stop offset="100%" style="stop-color:#FFFFFF"/></linearGradient><linearGradient x1="0 ...
```
I guess you removed the prefixes of the event names for brevity and that might be fine, but the text explaining the graph must then also be updated accordingly. That is true not only for this graph but for all.

d-stahl-ericsson replied:
Could you elaborate? The prefixes and suffixes haven't been in the diagrams for ages (if ever?). I assume you're referring to Eiffel...Event?

e-backmark-ericsson replied:
In the text on build-avoidance the events are referred to as 'CDefC1' and 'ArtCC1' for example. The same names appear in the circles in the current diagrams. In this PR you've remove the event name shortcut prefixes leaving only 'C1' and 'C1' which is not very distinct and clear.

e-backmark-ericsson commented:
I think this is a good improvement. The png files are not be needed anymore, since github can display the svg files natively, right? If the png files should be kept they should also be updated.
Btw, what is your svg editor of choice?

d-stahl-ericsson commented:
Technically the PNGs aren't needed, but we may as well keep them. PNGs are more universal than SVGs. Unless you want to be consistent in which tool is used to render them, why update them? There's no change in the visuals, so I don't see the point.

I find draw.io to satisfy my needs. If you prefer a desktop program, Inkscape is mature and popular.

e-backmark-ericsson commented:
Today the PNGs are rendered by Gliffy, from the .gliffy source files. The PNGs are the ones shown on the docs pages (linked within .md files). I suggest that we change those links to instead point to the .svg files within this PR and remove the PNG files rendered from the outdated .gliffy files. As commented on build-avoidance.svg it does not match the event names used in the current .gliffy files (and their generated .png files).

d-stahl-ericsson commented:
You're right. Clearly the translation wasn't as smooth as I first thought. Manual editing ensues...

By the way, regarding tooling. draw.io can only write SVGs. It imports them as static images :(

d-stahl-ericsson commented:
I fixed all errors I could detect in the SVGs, removed the PNGs and updated the image links in .md files. I found one case where the PNG and the .gliffy were inconsistent, so clearly getting rid of the PNGs altogether is a good thing.

e-backmark-ericsson commented:
Similar typo on confidence-level-joining.svg (but this one is inherited from the .gliffy file...) - Two events are called TCS3. The lower one should be TCS4.

e-backmark-ericsson commented on usage-examples/reference-data-sets/default.svg:
```diff
@@ -0,0 +1,5448 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<svg
+   xmlns:xhtml="http://www.w3.org/1999/xhtml"
+   xmlns:dc="http://purl.org/dc/elements/1.1/"
```
EDef1, EDef2 and ArtC3 are missing there event names (EnvironmentDefined and ArtifactCreated, respectively)
TCS3 exists in two events, the lower one should be TCS4

e-backmark-ericsson commented on usage-examples/test-execution.md:
```diff
@@ -52,13 +52,5 @@ To signal that a test case is about to be executed an [EiffelTestCaseTriggeredEv
 ## Test Activity Execution Implementation
+The [Eiffel Sepia architecture description](https://eiffel-community.github.io/eiffel-sepia/index.html) provides [a possible sequence diagram of roles interacting](https://eiffel-community.github.io/eiffel-sepia/test-orchestration.html) to realize the execution and reporting of test cases.
 To realize the event graph in this example, a sequence like the following is needed:
-![alt text](./test-execution-sequence.png "Sequence of Calls and Events of Test Execution Example")
```
I'd prefer that the sequence diagram was kept in this PR. I agree that the Sepia link could possibly replace it but that is a different story and more stuff on this page needs to be updated or migrated to Sepia before that can be done. The third section in the Introduction on this page depends on the sequence diagram being here for example.

e-backmark-ericsson commented:
delivery-interface.svg
'BASE_CHANGE' should be 'BASE, CHANGE'

test-execution-sequence.svg
All multiline strings are cropped.

And that's all for this review. Go ahead and update :)

e-backmark-ericsson commented:
Oh, sorry, one more thing. The PNGs are still there in the latest pushed commit.

d-stahl-ericsson commented:
How embarrassing :( I fixed the .svg typos and errors inherited from the .gliffy files. I also removed the text-execution-sequence.svg and restored the .png file. If we're going to keep it for now, we might as well keep it in its original .png format. If we do any additional work to it, we should refer to Sepia, so I didn't restore the .gliffy.

d-stahl-ericsson commented:
> Oh, sorry, one more thing. The PNGs are still there in the latest pushed commit.

Really? I'm pretty sure they're deleted. Check https://github.com/eiffel-community/eiffel/tree/7e8e4d4a19356d9f3dcd8e92abac4da140c1aad1/usage-examples

e-backmark-ericsson commented:
I'm satisfied now :)

:+1:

d-stahl-ericsson commented:
Awesome! You're a great reviewer and I truly appreciate your meticulousness. If nobody else has reviewed by Friday I'll merge, and then rebase #195.
